model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt

```diff
@@ -98,24 +109,37 @@ class RepositoryOverview(private val repoManager: IRepositoriesManager) {
     }
 }
 
-fun FlowOrInteractiveOrPhrasingContent.buildHistoryLink(repositoryId: String, branchName: String) {
+internal fun FlowOrInteractiveOrPhrasingContent.buildHistoryLink(repositoryId: String, branchName: String) {
     a("../history/${repositoryId.encodeURLPathPart()}/${branchName.encodeURLPathPart()}/") {
         +"Show History"
     }
 }
 
-fun FlowOrInteractiveOrPhrasingContent.buildExploreLatestLink(repositoryId: String, branchName: String) {
+internal fun FlowOrInteractiveOrPhrasingContent.buildExploreLatestLink(repositoryId: String, branchName: String) {
     a("../content/repositories/${repositoryId.encodeURLPathPart()}/branches/${branchName.encodeURLPathPart()}/latest/") {
         +"Explore Latest Version"
     }
 }
 
-fun FlowContent.buildDeleteForm(repositoryId: String) {
+internal fun FlowContent.buildDeleteRepositoryForm(repositoryId: String) {
     form {
         postButton {
             name = "delete"
             formAction = "../v2/repositories/${repositoryId.encodeURLPathPart()}/delete"
+            onClick = "return confirm('Are you sure you want to delete the repository $repositoryId?')"
             +"Delete Repository"
         }
     }
 }
+
+internal fun FlowContent.buildDeleteBranchForm(repositoryId: String, branchName: String) {
+    if (branchName == RepositoryId.DEFAULT_BRANCH) return
+    form {
+        deleteButton {
+            name = "delete"
+            formAction = "../v2/repositories/${repositoryId.encodeURLPathPart()}/branches/${branchName.encodeURLPathPart()}"
```

Comment:
How does it work that the browser doesn't end up in the response of the REST call after submission?

Reply:
fixed it with some javascript